### PR TITLE
Inactive tabs changes

### PR DESCRIPTION
**Pull request by MozillaNoah**
#21319

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/tabstray/browser/NormalBrowserTrayList.kt [hunk 1]
```diff
@@ -41,34 +44,41 @@ class NormalBrowserTrayList @JvmOverloads constructor(
 
     override val tabsFeature by lazy {
         val tabsAdapter = concatAdapter.browserAdapter
+        val inactiveTabsEnabled = context.settings().inactiveTabsAreEnabled
+        val tabFilter: (TabSessionState) -> Boolean = {
+            when {
+                FeatureFlags.tabGroupFeature && inactiveTabsEnabled ->
```
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/tabstray/browser/NormalBrowserTrayList.kt:50]: Do we need to check the feature flag if `inactiveTabsEnabled` is now a `featureFlagPreference`?

Comment (MozillaNoah) [reply app/src/main/java/org/mozilla/fenix/tabstray/browser/NormalBrowserTrayList.kt:50]: This is the FeatureFlag for tab groups not inactive tabs

Comment (rocketsroger) [reply app/src/main/java/org/mozilla/fenix/tabstray/browser/NormalBrowserTrayList.kt:50]: Ah, forgot about that.  Then this make sense.  Thanks

File: app/src/main/java/org/mozilla/fenix/tabstray/viewholders/NormalBrowserPageViewHolder.kt [hunk 2]
```diff
@@ -79,11 +79,12 @@ class NormalBrowserPageViewHolder(
         val browserAdapter = concatAdapter.browserAdapter
         val inactiveTabAdapter = concatAdapter.inactiveTabsAdapter
         val tabGroupAdapter = concatAdapter.tabGroupAdapter
+        val inactiveTabsAreEnabled = containerView.context.settings().inactiveTabsAreEnabled
 
         val selectedTab = browserStore.state.selectedNormalTab ?: return
 
         // Update tabs into the inactive adapter.
-        if (FeatureFlags.inactiveTabs && selectedTab.isNormalTabInactive(maxActiveTime)) {
+        if (FeatureFlags.inactiveTabs && selectedTab.isNormalTabInactive(maxActiveTime) && inactiveTabsAreEnabled) {
```
Comment (rocketsroger) [inline app/src/main/java/org/mozilla/fenix/tabstray/viewholders/NormalBrowserPageViewHolder.kt:87]: The check for the flag here is not needed.

Comment (MozillaNoah) [reply app/src/main/java/org/mozilla/fenix/tabstray/viewholders/NormalBrowserPageViewHolder.kt:87]: Good catch

**Review by rocketsroger — APPROVED** [review]
🚢

**Pull request merged**